x-pack/plugins/enterprise_search/public/applications/app_search/components/curations/curation/curation_logic.test.ts

```diff
@@ -0,0 +1,169 @@
+  const { http } = mockHttpValues;
+  const { navigateToUrl } = mockKibanaValues;
+  const { clearFlashMessages, flashAPIErrors } = mockFlashMessageHelpers;
+
+  const MOCK_CURATION_RESPONSE = {
+    id: 'cur-123456789',
+    last_updated: 'some timestamp',
+    queries: ['some search'],
+    promoted: [{ id: 'some-promoted-document' }],
+    organic: [
+      {
+        id: { raw: 'some-organic-document', snippet: null },
+        _meta: { id: 'some-organic-document', engine: 'some-engine' },
+      },
+    ],
+    hidden: [{ id: 'some-hidden-document' }],
+  };
+
+  const DEFAULT_VALUES = {
+    dataLoading: true,
+    curation: {
+      id: '',
+      last_updated: '',
+      queries: [],
+      promoted: [],
+      organic: [],
+      hidden: [],
+    },
+  };
+
+  beforeEach(() => {
+    jest.clearAllMocks();
+  });
+
+  it('has expected default values', () => {
+    mount();
+    expect(CurationLogic.values).toEqual(DEFAULT_VALUES);
+  });
+
+  describe('actions', () => {
+    describe('onCurationLoad', () => {
+      it('should set curation state & dataLoading to false', () => {
+        mount();
+
+        CurationLogic.actions.onCurationLoad(MOCK_CURATION_RESPONSE);
+
+        expect(CurationLogic.values).toEqual({
+          ...DEFAULT_VALUES,
+          curation: MOCK_CURATION_RESPONSE,
+          dataLoading: false,
+        });
+      });
+    });
+  });
+
+  describe('listeners', () => {
+    describe('loadCuration', () => {
+      it('should set dataLoading state', () => {
+        mount({ dataLoading: false }, { curationId: 'cur-123456789' });
+
+        CurationLogic.actions.loadCuration();
+
+        expect(CurationLogic.values).toEqual({
+          ...DEFAULT_VALUES,
+          dataLoading: true,
+        });
+      });
+
+      it('should make an API call and set curation state', async () => {
+        http.get.mockReturnValueOnce(Promise.resolve(MOCK_CURATION_RESPONSE));
+        mount({}, { curationId: 'cur-123456789' });
+        jest.spyOn(CurationLogic.actions, 'onCurationLoad');
+
+        CurationLogic.actions.loadCuration();
+        await nextTick();
+
+        expect(http.get).toHaveBeenCalledWith(
+          '/api/app_search/engines/some-engine/curations/cur-123456789'
+        );
+        expect(CurationLogic.actions.onCurationLoad).toHaveBeenCalledWith(MOCK_CURATION_RESPONSE);
+      });
+
+      it('handles errors/404s with a redirect to the Curations view', async () => {
+        http.get.mockReturnValueOnce(Promise.reject('error'));
+        mount({}, { curationId: 'cur-404' });
+
+        CurationLogic.actions.loadCuration();
+        await nextTick();
+
+        expect(flashAPIErrors).toHaveBeenCalledWith('error', { isQueued: true });
+        expect(navigateToUrl).toHaveBeenCalledWith('/engines/some-engine/curations');
+      });
+    });
+
+    describe('updateCuration', () => {
+      beforeAll(() => jest.useFakeTimers());
+      afterAll(() => jest.useRealTimers());
```

Comment:
Can you explain this bit? I don't think I've seen it in this codebase yet.

Reply:
Yes! This is my first time using `breakpoint` too so I just pinged Jason about this in Slack to clarify how it worked. I'll tag you in that thread here in a bit.

A bit more context:
- Jason introduced this way of working around `breakpoint` in https://github.com/elastic/kibana/pull/89636/files (I believe)
- [RelevanceTuningLogic](https://github.com/elastic/kibana/blob/6ab3fc010793e619dab1a83e3a80e435fb5c9090/x-pack/plugins/enterprise_search/public/applications/app_search/components/relevance_tuning/relevance_tuning_logic.test.ts#L310) also uses `breakpoint` and this method of skipping the breakpoint (which is where I learned & copied this from)
- General Jest timer mock documentation can be found here for more info: https://jestjs.io/docs/timer-mocks

Reply:
Generally speaking, any time in jest that you need to wait some length of time (whether it's setTimeout, debounce, or breakpoint), fake timers are what you should reach for. Using "fake" timers means you don't actually have to wait the specified length of time, which can slow down your test suite.